fix: added logout user before sso pipeline starts

Akanshu-2u opened the pull request:
### Description:
Users logging in through OAuth SSO could retain session data from previous logins, creating potential security risks and confusion when switching between accounts.

### Solution:
Added `logout(request)` call in `EdXOAuth2.start()` method to clear existing sessions before OAuth authentication begins.

### JIRA:
[BOMS-3](https://2u-internal.atlassian.net/browse/BOMS-3)

Akanshu-2u commented on auth_backends/backends.py:
```diff
@@ -2,9 +2,27 @@
+
+# .. toggle_name: ENABLE_OAUTH_SESSION_CLEANUP
+# .. toggle_implementation: SettingToggle
+# .. toggle_default: True
```
Changed default toggle value to False and further no test updates are needed. Thanks.

robrap replied:
Please resolve this comment. Thanks @Akanshu-2u.

Akanshu-2u commented:
> @Akanshu-2u changes look good, can you bump the version of the library so that we can do a release once we merge this?

Addressed. Added version bump of the library. Thank you.

robrap commented:
Thanks @feanil.
